src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/ResourceLocator.js

```diff
@@ -89,6 +101,13 @@ class ResourceLocator extends React.Component<FieldTypeProps<?string>> {
         onFinish();
     };
 
+    handleButtonClick = () => {
+        const {onFinish} = this.props;
+        this.forceRefresh = true;
```

Comment:
Why the indirection by using a separate `forceRefresh` variable? I would just extract the part generating the URL from the `addFinishHandler` above into a separate function, and call this function here.

Reply:
I now seperated this into a new function, but I transformed "generateUrl" in its own function to keep it clean.